### PR TITLE
Add new STM32F4 and measured boot example

### PR DESCRIPTION
**Pull request by tomoveu**
This PR has ~~three~~ two goals:
- Add wolfBoot-example for STM32F4
- Add measured boot example
- ~~Use TLS1.3 to communicate over the internet~~

This example is based on the existing test-app in wolfBoot and:
- It uses STM32F4-Discovery 
- It uses TPM SLB9670
- It uses UART
- ~~It has a HTTPS client~~

This is a simple way to demonstrate the log(trace) that measured boot creates for the user application.

Signed-off-by: Dimitar Tomov <dimi@wolfssl.com>

### CODE REVIEW

**Comment by tomoveu** [comment]
Quick update: The STM32F4BB Baseboard is no longer in manufacturing, therefore I ordered the STM32F4WIFI extension board. Using WiFi would require me to add extra code(driver), but will make the example use currently available STM32 hardware, so everyone could use it :)

**Comment by tomoveu** [comment]
@danielinux This is ready for testing.

Occasionally, I get "isr_fault" due to the jump wires between my STM32 and TPM. I want to solder my TPM like you have done.

Make sure the wolfTPM submodule in wolfBoot is fetched as well. This is the 1st wolfboot-example that requires wolfTPM 💡  

Example README here, UART pinout, TPM pinout - https://github.com/tomoveu/wolfBoot-examples/tree/add-stm32f4-example/test-app-STM32F4-measured-boot#compiling

File: test-app-STM32F4-measured-boot/src/app_stm32f4.c [hunk 1]
```diff
@@ -0,0 +1,404 @@
+/* stm32f4.c
```
Comment (dgarske) [inline test-app-STM32F4-measured-boot/src/app_stm32f4.c]: Please update the filename and description.

File: test-app-STM32F4-measured-boot/src/app_stm32f4.c [hunk 2]
```diff
@@ -0,0 +1,404 @@
...
+static int app_tpm2_IoCb(TPM2_CTX* ctx, const byte* txBuf, byte* rxBuf,
+    word16 xferSz, void* userCtx)
+{
+    (void)userCtx;
+    (void)ctx;
+    word16 i;
+
+    spi_cs_on(SPI_CS_TPM);
+
+    memset(rxBuf, 0, xferSz);
```
Comment (dgarske) [inline test-app-STM32F4-measured-boot/src/app_stm32f4.c]: Consider using XMEMSET for consistency.

File: test-app-STM32F4-measured-boot/src/app_stm32f4.c [hunk 3]
```diff
@@ -0,0 +1,404 @@
...
+        XMEMCPY(digest, pcrReadResp.pcrValues.digests[0].buffer,
+                pcrReadResp.pcrValues.digests[0].size);
+        rc = 0;
+    }
+
+    return rc;
+}
+
+volatile uint32_t time_elapsed = 0;
+volatile uint32_t testme = 1;
```
Comment (dgarske) [inline test-app-STM32F4-measured-boot/src/app_stm32f4.c]: I prefer globals at the top of file. Also if not being extern'd please make static.

File: test-app-STM32F4-measured-boot/src/app_stm32f4.c [hunk 4]
```diff
@@ -0,0 +1,404 @@
...
+void main(void) {
```
Comment (dgarske) [inline test-app-STM32F4-measured-boot/src/app_stm32f4.c]: Brace on new line for functions.

**Review by dgarske — CHANGES_REQUESTED** [review]
Just a few minor things at first glance. I plan to setup and test this tomorrow.

File: test-app-STM32F4-measured-boot/src/app_stm32f4.c (hunk 4 above)
Comment (tomoveu) [reply test-app-STM32F4-measured-boot/src/app_stm32f4.c]: Typo. Thanks for catching it.

File: test-app-STM32F4-measured-boot/src/app_stm32f4.c (hunk 3 above)
Comment (tomoveu) [reply test-app-STM32F4-measured-boot/src/app_stm32f4.c]: Moved and made static.

File: test-app-STM32F4-measured-boot/src/app_stm32f4.c (hunk 2 above)
Comment (tomoveu) [reply test-app-STM32F4-measured-boot/src/app_stm32f4.c]: Will do.

File: test-app-STM32F4-measured-boot/src/app_stm32f4.c (hunk 1 above)
Comment (tomoveu) [reply test-app-STM32F4-measured-boot/src/app_stm32f4.c]: Thanks.

**Comment by tomoveu** [comment]
@dgarske thank you for the review 🤝 

I have made the requested changes. I also extended the README file and added my openocd config.

Please make sure the wolfTPM submodule in wolfBoot is fetched as well, because this is the first wolfboot-example that requires wolfTPM 😃

**Review by dgarske — APPROVED** [review]
Over to @danielinux for final. Consider merging PR https://github.com/wolfSSL/wolfBoot/pull/103 and updating the wolfBoot example submodule.

**Review by danielinux — CHANGES_REQUESTED** [review]
This looks great. I only have one request. Can the '.bmp' image in the README be in a compressed format (e.g. png) to reduce size?

**Comment by tomoveu** [comment]
Sure. I will convert it and force-push to have clean git history

**Review by danielinux — APPROVED** [review]

**Pull request merged**